internal/restic/snapshot_find.go

```diff
@@ -71,6 +80,58 @@ func FindLatestSnapshot(ctx context.Context, repo Repository, targets []string,
 	return latestID, nil
 }
 
+// FindLatestSnapshotBefore finds latest snapshot with optional target/directory, tags and hostname filters.
+func FindLatestSnapshotBefore(ctx context.Context, repo Repository, targets []string, hostname string, timeStamp time.Time) (ID, error) {
+	var err error
+	absTargets, err := resolveTargets(targets)
+	if err != nil {
+		return ID{}, err
+	}
+
+	var (
+		latest   time.Time
+		latestID ID
+		found    bool
+	)
+
+	err = ForAllSnapshots(ctx, repo, nil, func(id ID, snapshot *Snapshot, err error) error {
+		if err != nil {
+			return errors.Errorf("Error loading snapshot %v: %v", id.Str(), err)
+		}
+
+		if snapshot.Time.After(timeStamp) {
```

Comment:
What about using the zero-value of `time.Time(0)` to disable the time check? Then `FindLatestSnapshot` and `FindLatestSnapshotBefore` can internally be implemented using the same function.

Reply:
Combined the two functions. Used a `*time.Time` instead of the zero value to make it more clear this is an optional argument since there's no constant for the max `time.Time` and passing `time.Time(0)` as the max snapshot timestamp to disable the behavior seems a bit error-prone. Let me know if you still would prefer `time.Time(0)`.